README.md

```diff
@@ -54,7 +57,165 @@ with an appropriate schema.
 
 ## Development
 
-You'll need to publish local artifacts for VectorPipe, GeoTrellis and GeoMesa.
+Development of OSMesa happens via Docker containers run inside of a Vagrant machine.
+While this is a complex setup, it allows us full control over the development environment,
+and how it maps to the deployment environment.
+
+__Note__: If you might have permissions issues with Docker writing to the host machine
+(some OS's do), you might want to remove the `config.vm.synced_folder` lines in the
+`Vagrantfile` and simply copy up credentials to the VM/let the VM have it's own Ivy cache, etc.
+
+### Requirements
+
+* Vagrant 1.8+
+* VirtualBox 4.3
+* Ansible 2.1+
+
+### Getting Started
+
+The first steps are to build some unpublished binaries:
+
+#### Build GeoMesa and make hbase-dist binaries available
+
+First thing to is to [build and install GeoMesa](https://github.com/locationtech/geomesa#building-from-source).
+You can run `mvn clean install -T8 -am -DskipTests` on a clone of GeoMesa to accomplish this.
+
+_Note:__ - Until https://github.com/locationtech/geomesa/pull/1751 is merged, you'll have to patch that into master.
+
+__IMPORTANT__: Next, move the `geomesa-hbase-dist` installed in your local maven repository. Move it to both these locations:
+
+- `services/hbase/geomesa-hbase-dist.tar.gz`
+- `services/geoserver/geomesa-hbase-dist.tar.gz`
+
+For instance, running this if you built GeoMesa and installed locally:
+- `cp ~/.m2/repository/org/locationtech/geomesa/geomesa-hbase-dist_2.11/1.4.0-SNAPSHOT/geomesa-hbase-dist_2.11-1.4.0-SNAPSHOT-bin.tar.gz services/hbase/geomesa-hbase-dist.tar.gz`
+- `cp ~/.m2/repository/org/locationtech/geomesa/geomesa-hbase-dist_2.11/1.4.0-SNAPSHOT/geomesa-hbase-dist_2.11-1.4.0-SNAPSHOT-bin.tar.gz services/geoserver/geomesa-hbase-dist.tar.gz`
+
+
+You'll also need the local maven repository to build the scala code in this project; this means that if you
+have Vagrant syncing your `~/.m2` folder, you can build directly on your machine, but if
+you have disabled that syncing you'll have to build and install it inside of the Vagrant box.
+
+#### Build the latest GeoTrellis
+
+The same thing as above applies here: if Vagrant is syncing to the host `~/.ivy2`, you can publish
+local binaries on the host machine; otherwise you'll have to publish-local from the Vagrant box.
+
+- Clone https://github.com/locationtech/geotrellis
+- `cd geotrellis`
+- `scripts/publish-local.sh`
+
+This one requires that SBT be locally installed.
+
+_TODO:_ Should we bake all these in a docker container to make it easier?
+
+#### Build the latest VectorPipe
+
+- Clone https://github.com/geotrellis/vectorpipe
+- `cd vectorpipe`
+- `sbt publish-local`
```

Comment:
Hit a snag here that I haven't been able to work around:

```
$ sbt publish-local
[info] Loading global plugins from /Users/hcastro/.sbt/0.13/plugins
[info] Loading project definition from /Users/hcastro/Projects/vectorpipe/project
/Users/hcastro/Projects/vectorpipe/ensime.sbt:1: error: not found: value ensimeIgnoreMissingDirectories
ensimeIgnoreMissingDirectories in ThisBuild := true
^
[error] Type error in expression
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? q
$ brew info sbt
sbt: stable 1.0.2
Build tool for Scala projects
http://www.scala-sbt.org
/usr/local/Cellar/sbt/1.0.2 (482 files, 61.0MB) *
  Built from source on 2017-09-20 at 08:41:51
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/sbt.rb
==> Requirements
Required: java >= 1.8 ✔
==> Caveats
You can use $SBT_OPTS to pass additional JVM options to SBT:
   SBT_OPTS="-XX:+CMSClassUnloadingEnabled -XX:MaxPermSize=256M"

This formula uses the standard Lightbend sbt launcher script.
Project specific options should be placed in .sbtopts in the root of your project.
Global settings should be placed in /usr/local/etc/sbtopts
```

Reply:
Is this an ensime specific problem? @fosskers might know

Reply:
Ah, looking at the stack trace, that error is actually coming from your local ensibe.sbt...

I'll make sure things can be built in the VM at least and push that up.

Reply:
This is an ensime issue - if you add `addSbtPlugin("org.ensime" % "sbt-ensime" % "2.0.1")` to your ~/.sbt/0.13/plugins/plugins.sbt, things should work

Reply:
I should have been clearer. I think this is caused by the specific version of ensime I mentioned not being available.

Reply:
I added a `scripts/publish-local.sh all` that should publish everything you need, and updated the README. I don't have a good connection right now so I can't fully test.